components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -97,27 +98,20 @@ class ToolbarPresenter(
 
         extensions.forEach { extension ->
             extension.browserAction?.let { extensionAction ->
-                val toolbarAction = convertToToolbarAction(extensionAction)
-                // We need to verify, if we this toolbarAction already exits on the toolbar
-                // if so, we don't need to re-add it, just replace it with the new one.
-                // https://github.com/mozilla-mobile/android-components/issues/4651
-                toolbar.addBrowserAction(toolbarAction)
+                val toolbarAction = WebExtensionToolbarAction(
+                    browserAction = extensionAction,
+                    listener = extensionAction.onClick)
+
+                if (!toolbarActionSet.contains(toolbarAction)) {
```

Comment:
If we use the map approach, when we notice that the item is already there we need to replace its `WebExtensionToolbarAction.browserAction` references with the new one (`extensionAction`) other wise we are going to added to the toolbar.